lib/compat/wordpress-6.0/block-editor-settings.php

```diff
@@ -0,0 +1,165 @@
+		gutenberg_is_edit_site_page( get_current_screen()->id )
+	) {
+		$context = 'site-editor';
+	}
+
+	if (
+		defined( 'REST_REQUEST' ) &&
+		REST_REQUEST &&
+		isset( $_GET['context'] ) &&
+		'mobile' === $_GET['context']
+	) {
+		$context = 'mobile';
+	}
+
+	if ( 'other' === $context ) {
+		// Make sure the styles array exists.
+		// In some contexts, like the navigation editor, it doesn't.
+		if ( ! isset( $settings['styles'] ) ) {
+			$settings['styles'] = array();
+		}
+
+		$styles_without_existing_global_styles = array();
+		foreach ( $settings['styles'] as $style ) {
+			if (
+				! isset( $style['__unstableType'] ) ||
+				// '__unstableType' is 'globalStyles' for WordPress 5.8 and 'presets' for WordPress 5.9.
+				//
+				// Note that styles classified as'theme', can be from the theme stylesheet
+				// or from the theme.json (the styles section).
+				// We are unable to identify which is which, so we can't remove and recreate those.
+				// Instead, we reload the theme.json styles from the plugin.
+				// Because they'll use the same selectors and load later,
+				// they'll have higher priority than core's.
+				//
+				// Theoretically, this approach with 'theme' styles could be problematic:
+				// if we remove style properties in the plugin, if selectors change, etc.
+				// We need to address this issue directly in core by alowing to identify
+				// styles coming from theme.json.
+				//
+				// A final note about 'theme' styles: this flag is used to identify theme
+				// styles that may need to be removed if the user toggles
+				// "Preferences > Use theme styles" in the preferences modal.
+				//
+				! in_array( $style['__unstableType'], array( 'globalStyles', 'presets' ), true )
+			) {
+				$styles_without_existing_global_styles[] = $style;
+			}
+		}
+
+		$new_global_styles = array();
+		$presets           = array(
+			array(
+				'css'            => 'variables',
+				'__unstableType' => 'presets',
+			),
+			array(
+				'css'            => 'presets',
+				'__unstableType' => 'presets',
+			),
+		);
+		foreach ( $presets as $preset_style ) {
+			$actual_css = gutenberg_get_global_stylesheet( array( $preset_style['css'] ) );
+			if ( '' !== $actual_css ) {
+				$preset_style['css'] = $actual_css;
+				$new_global_styles[] = $preset_style;
+			}
+		}
+
+		if ( WP_Theme_JSON_Resolver::theme_has_support() ) {
+			$block_classes = array(
+				'css'            => 'styles',
+				'__unstableType' => 'theme',
+			);
+			$actual_css    = gutenberg_get_global_stylesheet( array( $block_classes['css'] ) );
+			if ( '' !== $actual_css ) {
+				$block_classes['css'] = $actual_css;
+				$new_global_styles[]  = $block_classes;
+			}
+		}
+
+		$settings['styles'] = array_merge( $new_global_styles, $styles_without_existing_global_styles );
+	}
+
+	// Copied from get_block_editor_settings() at wordpress-develop/block-editor.php.
+	$settings['__experimentalFeatures'] = gutenberg_get_global_settings();
```

Comment:
We've done some modifications to the data available in theme.json that we expect to land in 6.0. While this doesn't have any changes with respect to what's in WordPress core, we still need to make sure this calls the `gutenberg_*` methods so the said modifications are loaded in the editors.